diff --git a/dist/cjs/Match.cjs b/dist/cjs/Match.cjs
index a3bcc0c5b9dbfc0048388a90f9fe8fab9bb76dc5..ed3cbfcce22d3f92c25c1aa7ce6549602aa0b176 100644
--- a/dist/cjs/Match.cjs
+++ b/dist/cjs/Match.cjs
@@ -106,8 +106,10 @@ var MatchInner = react.memo(function MatchInnerImpl({ match }) {
 	]);
 	if (match.status === "pending") {
 		if (router.ssr && !canWrapInSuspense(router, route, match.ssr)) return out;
-		if (router._tx) throw router._tx[5];
-		return renderPending(router, route);
+		if (router._tx) throw router._tx[5] ?? new Promise(() => {});
+		const pending = renderPending(router, route);
+		if (!pending) throw new Promise(() => {});
+		return pending;
 	}
 	if (match.status === "notFound") return require_renderRouteNotFound.renderRouteNotFound(router, route, match.error);
 	if (match.status === "error") {
diff --git a/dist/esm/Match.js b/dist/esm/Match.js
index 61bdaf0190e7067af284f4fbb24a4d03b0a25003..db62456447de3e8943a440be3eab353c29842c58 100644
--- a/dist/esm/Match.js
+++ b/dist/esm/Match.js
@@ -104,8 +104,10 @@ var MatchInner = React$1.memo(function MatchInnerImpl({ match }) {
 	]);
 	if (match.status === "pending") {
 		if (router.ssr && !canWrapInSuspense(router, route, match.ssr)) return out;
-		if (router._tx) throw router._tx[5];
-		return renderPending(router, route);
+		if (router._tx) throw router._tx[5] ?? new Promise(() => {});
+		const pending = renderPending(router, route);
+		if (!pending) throw new Promise(() => {});
+		return pending;
 	}
 	if (match.status === "notFound") return renderRouteNotFound(router, route, match.error);
 	if (match.status === "error") {
