diff --git a/dist/constants.cjs b/dist/constants.cjs
index 2155a64ace7ad61892f0fc86e887b3ed4a8fe75f..0f6bf45eccd5d37e7539d0ad9fdd2527b7dc444e 100644
--- a/dist/constants.cjs
+++ b/dist/constants.cjs
@@ -384,6 +384,7 @@ function isSingleLetterKey(key) {
 * ```
 */
 function normalizeKeyName(key) {
+	if (typeof key !== "string") return "";
 	if (!key) return "";
 	if (key in KEY_ALIASES) return KEY_ALIASES[key];
 	if (isSingleLetterKey(key)) {
diff --git a/dist/constants.js b/dist/constants.js
index b05b01ed7e0b618a950cc6e465d9c9624ffdce65..2b6928a8e72ed1af67ca70db0523bec475116da9 100644
--- a/dist/constants.js
+++ b/dist/constants.js
@@ -384,6 +384,7 @@ function isSingleLetterKey(key) {
 * ```
 */
 function normalizeKeyName(key) {
+	if (typeof key !== "string") return "";
 	if (!key) return "";
 	if (key in KEY_ALIASES) return KEY_ALIASES[key];
 	if (isSingleLetterKey(key)) {
